src/test/java/io/horrorshow/codey/time/DiscordTimezoneTest.java

```diff
@@ -0,0 +1,78 @@
+package io.horrorshow.codey.time;
+
+import io.horrorshow.codey.discordutil.CodeyConfig;
+import io.horrorshow.codey.discordutil.DiscordMessage;
+import io.horrorshow.codey.discordutil.DiscordUtils;
+import io.horrorshow.codey.formatter.DiscordCodeFormatter;
+import io.horrorshow.codey.formatter.JavaFormatter;
+import net.dv8tion.jda.api.JDA;
+import net.dv8tion.jda.api.entities.Message;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import java.time.DayOfWeek;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.mock;
+
+class DiscordTimezoneTest {
+    @Mock
+    JDA jda;
+    CodeyConfig codeyConfig;
+    DiscordUtils utils;
+    DiscordTimezone discordTimezone;
+
+    @BeforeEach
+    void init() {
+        MockitoAnnotations.openMocks(this);
+        codeyConfig = new CodeyConfig();
+        utils = new DiscordUtils(jda, codeyConfig);
+        discordTimezone = new DiscordTimezone(jda, utils);
+    }
+
+    @Test
+    void test_no_weekday()
+    {
+        String[] tests = {
+                "18:00 Europe/Amsterdam", // match
+                "15:011 America/Chicago", // no match
+                "12:35 Am/Chic", // no match
+                "11:35 what are you doing?" // no match
+        };
+
+        assertEquals(discordTimezone.matchTimes(tests[0])[0].parseTime()[0], 18);
```

Comment:
you can see here that using array indices instead of attributes makes things very cryptic. How would someone know that the int at `[0]` is the hour, at `[1]` it's the minute.